### PR TITLE
Restoring the selection

### PR DESCRIPTION
**Pull request by maxGimeno**
Restores the selection feature, lost with the enhancing of the axis system.

### CODE REVIEW

**Comment by sloriot** [comment]
Merged in CGAL-4.8-Ic-109

File: Polyhedron/demo/Polyhedron/Viewer.cpp [hunk 1]
```diff
@@ -109,7 +111,12 @@ bool Viewer::inFastDrawing() const
 void Viewer::draw()
 {
   glEnable(GL_DEPTH_TEST);
+  glClear(GL_DEPTH_BUFFER_BIT);
+  if(QOpenGLFramebufferObject::hasOpenGLFramebufferBlit())
+    saveDepthBuffer();
   d->draw_aux(false, this);
+ if(QOpenGLFramebufferObject::hasOpenGLFramebufferBlit())
+   restoreDepthBuffer();
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Viewer.cpp]: Indentation!

File: Polyhedron/demo/Polyhedron/Viewer.cpp [hunk 2]
```diff
@@ -1165,3 +1167,23 @@ void Viewer::wheelEvent(QWheelEvent* e)
...
 }
+
+void Viewer::saveDepthBuffer()
+{
+  if(depth_fbo)
+    delete depth_fbo;
+  int width = rect().width();
+  int height = rect().height();
+ depth_fbo = new QOpenGLFramebufferObject(QSize(width,height),QOpenGLFramebufferObject::Depth);
+ depth_fbo->release();
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Viewer.cpp]: Indentation

File: Three/include/CGAL/Three/Viewer_interface.h [hunk 3]
```diff
@@ -99,8 +99,12 @@ class VIEWER_EXPORT Viewer_interface : public QGLViewer, public QOpenGLFunctions
...
   //!Allows OpenGL 2.1 context to get access to glVertexAttribDivisor.
   PFNGLVERTEXATTRIBDIVISORARBPROC glVertexAttribDivisor;
-  //!Allows OpenGL 2.1 context to get access to gkFrameBufferTexture2D.
+  //!Allows OpenGL 2.1 context to get access to glFrameBufferTexture2D.
   PFNGLFRAMEBUFFERTEXTURE2DEXTPROC glFramebufferTexture2D;
+  enum filter {
+   LINEAR=0,
+   NEAREST
+  };
   //!@returns true if glVertexAttribDivisor, and glDrawArraysInstanced are found.
```
Comment (lrineau) [inline Three/include/CGAL/Three/Viewer_interface.h]: The enum `filter` seems unused!

Comment (maxGimeno) [inline Three/include/CGAL/Three/Viewer_interface.h]: The fix doesn't work anyway.

2016-01-26 15:16 GMT+01:00 Laurent Rineau <notifications@github.com>:

**Comment by maxGimeno** [comment]
Ready for merging

**Comment by sloriot** [comment]
Replaced by #691
#658 introduces bugs that can't be fixed according to @maxGimeno

**Pull request closed without merging**